akka-actor-typed-tests/src/test/scala/docs/akka/typed/StyleGuideDocExamples.scala

```diff
@@ -548,86 +551,26 @@ object StyleGuideDocExamples {
-        case Nil          => Behaviors.unhandled
-        case head :: tail => head.applyOrElse(command, handle(_, tail))
-      }
-    }
-    //#handle-partial
-
-    //#top-level-behaviors-partial
-    val zero: Behavior[Command] = Behaviors.receiveMessage { command =>
-      handle(command, List(getHandler(0)))
-    }
-
-    def nonZero(initialCapacity: Int): Behavior[Command] = Behaviors.receiveMessage { command =>
-      handle(command, List(getHandler(initialCapacity), setHandlerNotZero(initialCapacity)))
-    }
-    //#top-level-behaviors-partial
-  }
-
-  object BehaviorCompositionWithReceivePartial {
-
-    //#messages-sealed-composition
-    sealed trait Command
-    case object Down extends Command
-    final case class GetValue(replyTo: ActorRef[Value]) extends Command
-    final case class Value(n: Int)
-    //#messages-sealed-composition
-
-    //#get-handler
-    def getHandler(value: Int): Behavior[Command] = Behaviors.receiveMessagePartial {
-      case GetValue(replyTo) =>
-        replyTo ! Value(value)
-        Behaviors.same
-    }
-    //#get-handler
-
-    //#set-handler-non-zero
-    def setHandlerNotZero(value: Int): Behavior[Command] = Behaviors.receiveMessagePartial {
+    //#set-handler-zero-partial
+    def setHandlerZero(log: Logger): PartialFunction[Command, Behavior[Command]] = {
       case Down =>
-        if (value == 1)
-          zero
-        else
-          nonZero(value - 1)
-    }
-    //#set-handler-non-zero
-
-    //#set-handler-zero
-    val setHandlerZero: Behavior[Command] = Behaviors.receivePartial {
-      case (context, Down) =>
-        context.log.error("Counter is already at zero!")
+        log.error("Counter is already at zero!")
         Behaviors.same
     }
-    //#set-handler-zero
-
-    //#handle
-    @tailrec def handle(
-        command: Command,
-        handlers: List[Behavior[Command]],
-        ctx: ActorContext[Command]): Behavior[Command] = {
-      handlers match {
-        case Nil          => Behaviors.unhandled
-        case head :: tail =>
-          //Process the given message with the next Behavior in the list
-          val next = Behavior.interpretMessage(head, ctx, command)
-          if (Behavior.isUnhandled(next)) handle(command, tail, ctx)
-          else next
-      }
-    }
-    //#handle
+    //#set-handler-zero-partial
 
-    //#top-level-behaviors
+    //#top-level-behaviors-partial
     val zero: Behavior[Command] = Behaviors.receive { (context, command) =>
-      handle(command, List(getHandler(0), setHandlerZero), context)
+      getHandler(0).orElse(setHandlerZero(context.log))(command)
     }
 
-    def nonZero(initialCapacity: Int): Behavior[Command] = Behaviors.receive { (context, command) =>
-      handle(command, List(getHandler(initialCapacity), setHandlerNotZero(initialCapacity)), context)
+    def nonZero(capacity: Int): Behavior[Command] = Behaviors.receiveMessage { command =>
+      getHandler(capacity).orElse(setHandlerNotZero(capacity))(command)
```

Comment:
With `receivePartial` and `PartialFunction[(ActorContext[Command], Command), Behavior[Command]]` this could be even more streamlined _and_ deal with unhandled commands without crashing the actor, but would complicate signatures a little bit because of the context type param.

Reply:
In this particular case we know for sure that the composition of both `PartialFunction` produces a Total Function as there is no case unhandled, unfortunately we can tell that to the compiler (Or I don't know how).

Do you think that using `receivePartial` here might give the wrong intuition that there are cases unhandled?

Reply:
I was more thinking that we know that, but since it is not expressed with the type system, it may not hold true forever, while a `receivePartial` would live up to the partial-functioness even if messages are added.

Not super important though IMO.

Reply:
To avoid signature complication maybe I could do something like this:
```scala
val zero: Behavior[Command] = Behaviors.setup { context =>
  // Even though in this particular case we could use `receiveMessage` as we cover all 
  // cases, we use `receiveMessagePartial` instead to cover potential future unhandled message cases
  Behaviors.receiveMessagePartial(getHandler(0).orElse(setHandlerZero(context.log)))
}
```
Is that what you had in mind?

Reply:
No, but I think that looks a little bit nicer than the current.

I think the comment would fit better in the doc page text above or below the sample though.

Reply:
Pushed some changes, let me know how you like them.

And thanks for your time!

Reply:
Looks good 👍